adapters/repos/db/inverted/bm25_searcher.go

```diff
@@ -212,12 +201,13 @@ func (b *BM25Searcher) wand(
 						err = termErr
 						return
 					}
-					results[k] = termResult
-					indices[k] = docIndices
+					resultsLock.Lock()
```

Comment:
This is great, as it may solve an out-of-bounds that was reported in the community Slack.

Reply:
Regarding this, I have a suggestion: I'd split the BM25 searcher changes into a different PR that targets `stable/1.28`, and keep the new tokenizer changes in this PR targeting master as a feature for the `1.29` release

Reply:
I'd like to get the dictionary fix in quickly, since it isn't working in production (the docker build doesn't include the dictionaries in release)

Reply:
Do the BM25 searcher changes influence that, Anyway, we can still backport those changes to 1.28.